File: 初级算法/数组/两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        if(nums.size() <= 1)
            return result;
        vector<int>::iterator pos1 = nums.begin();
        vector<int>::iterator pos2 = nums.begin() + 1;
        while(pos1 != nums.end()){
            while(pos2 != nums.end()){
                if( *pos1 + *pos2 == target){
                    result.push_back(pos1-nums.begin());
                    result.push_back(pos2-nums.begin());
                    return result;
                }else
                    ++ pos2;
            }
            ++ pos1;
            pos2 = pos1 + 1;
        }
        return result;
    }
};

File: java/两数之和.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] elements = new int[2];
        int temp = 0;
        for(int i = 0; i < nums.length - 1 ; i++)
            for(int j = i+1; j < nums.length; j++){
                temp = nums[i] + nums[j];
                if(temp == target){
                    elements[0] = i;
                    elements[1] = j;
                }
            }
     return elements;   
    }
}